be/src/connector/hive_connector.cpp

```diff
@@ -576,16 +613,13 @@ Status HiveDataSource::_init_scanner(RuntimeState* state) {
     if (scan_range.__isset.paimon_deletion_file && !scan_range.paimon_deletion_file.path.empty()) {
         scanner_params.paimon_deletion_file = std::make_shared<TPaimonDeletionFile>(scan_range.paimon_deletion_file);
     }
-    scanner_params.use_datacache = _use_datacache;
-    scanner_params.enable_populate_datacache = _enable_populate_datacache;
-    scanner_params.enable_datacache_async_populate_mode = _enable_datacache_aync_populate_mode;
-    scanner_params.enable_datacache_io_adaptor = _enable_datacache_io_adaptor;
-    scanner_params.datacache_evict_probability = _datacache_evict_probability;
-    scanner_params.datacache_priority = _datacache_priority;
-    scanner_params.datacache_ttl_seconds = _datacache_ttl_seconds;
+
+    // setup options for datacache
+    scanner_params.datacache_options = _datacache_options;
```

Comment:
Why not pass the options by pointer to avoid copying all elements every time? Also, if the datacache is not used, the options is null, which is also more friendly to memory.